File: app/src/main/java/com/rachelleignacio/listbucket/presentation/presenters/ListBucketFragmentPresenter.java
package com.rachelleignacio.listbucket.presentation.presenters;

import android.support.v4.app.FragmentManager;

import com.rachelleignacio.listbucket.domain.models.List;

/**
 * Created by rachelleignacio on 3/31/17.
 */

public interface ListBucketFragmentPresenter {

    interface View {
        void showLists(java.util.List<List> lists);
        void onCLickList(List listToView);
        void onClickCreateList();
        void onListSwipedToDelete(int position);
        void onClickRenameList(List listToRename);
    }

    void getLists();
    void showCreateListDialog(FragmentManager fragmentManager);
    void deleteListFromBucket(int position);
    void showRenameListDialog(FragmentManager fragmentManager, List listToRename);
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/presentation/presenters/ListBucketFragmentPresenter.kt
package com.rachelleignacio.listbucket.presentation.presenters

import android.support.v4.app.FragmentManager
import com.rachelleignacio.listbucket.domain.models.List

/**
 * Created by rachelleignacio on 8/23/17.
 */
interface ListBucketFragmentPresenter {
    interface View {
        fun showLists(lists: MutableList<List>)
        fun onClickList(listToView: List)
        fun onClickCreateList()
        fun onListSwipedToDelete(adapterPosition: Int)
        fun onClickRenameList(listToRename: List)
    }

    fun getLists()
    fun showCreateListDialog(fragmentManager: FragmentManager)
    fun deleteListFromBucket(adapterPosition: Int)
    fun showRenameListDialog(fragmentManager: FragmentManager, listToRename: List)
}